src/replicache.ts

```diff
@@ -35,6 +35,9 @@ type MaybePromise<T> = T | Promise<T>;
 /** The key name to use in localStorage when synchronizing changes. */
 const storageKeyName = (name: string) => `/replicache/root/${name}`;
 
+/** The maximum number of time to call out to getDataLayerAuth before giving up and throwing an error. */
+const MAX_REAUTH_TRIES = 32;
```

Comment:
This can just be `1`, because even if we give up, we're just going to try again at the next sync interval. And there's no reason to expect that if auth failed explicitly (http 401) it's going to suddenly succeed if we retry synchronously.

Reply:
I don't think we want `1` here. The flow I'm imagining is:

- You get an auth error.
- getDataLayerAuth is called
  - This brings up a dialog asking for user credentials
  - Once that is done getDataLayerAuth resolves
- We try again

If we only try once and the user makes a typo when authenticating they do not get to try again.

Reply:
Ah, this was not the use case I was imagining for getAuthToken.

I was just imagining it was used for the token refresh dance that oauth-style clients must typically do.

For example, in Firebase, the token you have on the client side expires every hour. When it expires, client must obtain a new one by making a request to its own server, which has a different auth token called the "refresh token". The refresh token is very-long-lived, but is not exposed to clients for security reasons. The server uses the refresh token to get a new short-lived token and returns that to the client.

So the use case I was imagining was all completely automated, which is why retrying didn't make sense to me.

I had not considered this use case you were thinking about. Let me ponder it.